app/vep/vep_resources.py

```diff
@@ -70,21 +74,70 @@ async def submit_vep(request: Request):
 async def vep_status(request: Request, submission_id: str):
     try:
         workflow_status = await get_workflow_status(submission_id)
-        return JSONResponse(content=workflow_status.dict())
+        submission_status = PipelineStatus(
+            submission_id=submission_id, status=workflow_status
+        )
+        return JSONResponse(content=submission_status.dict())
 
     except HTTPError as http_error:
-        if http_error.response.status_code in [403,400]:
+        if http_error.response.status_code in [403, 400]:
             response_msg = json.dumps(
                 {
-                    "status_code": status.HTTP_404_NOT_FOUND,
                     "details": f"A submission with id {submission_id} was not found",
                 }
             )
             return PlainTextResponse(
                 response_msg, status_code=status.HTTP_404_NOT_FOUND
             )
 
-        return response_error_handler(result={"status": http_error.response.status_code})
+        return response_error_handler(
+            result={"status": http_error.response.status_code}
+        )
+    except Exception as e:
+        logging.debug(e)
+        return response_error_handler(result={"status": 500})
+
+
+@router.get("/submissions/{submission_id}/download", name="download_results")
+async def download_results(request: Request, submission_id: str):
+    try:
+        workflow_status = await get_workflow_status(submission_id)
+        submission_status = PipelineStatus(
+            submission_id=submission_id, status=workflow_status["workflow"]["status"]
+        )
+        if submission_status.status == "SUCCEEDED":
+            input_vcf_file = workflow_status["workflow"]["params"]["vcf"]
+            input_vcf_path = FilePath(input_vcf_file)
+            results_file_path = input_vcf_path.with_name("_VEP").with_suffix(".vcf.gz")
+            return FileResponse(
+                results_file_path,
+                media_type="application/gzip",
+                filename=results_file_path.name,
+            )
+        else:
+            response_msg = json.dumps(
+                {
+                    "details": f"A submission with id {submission_id} is not yet finished",
+                }
+            )
+            return PlainTextResponse(
```

Comment:
I think this should be a JSON response so it can produce a valid 404.

Reply:
This is now changed for both /download and /status endpoint  to maintain consistency.